### PR TITLE
feat: Improve inferred types from `getContracts`

### LINKED ISSUE #224: contract returned by getContract is of type Abi making it lose reference to the specific types of the contract.
The contract returned by `getContract` method is of type `import { type Abi } from 'viem'` making it lose reference to the specific types of the contract.

Ideally, when calling `getContract`, it. should be smart enough to infer the Abi type based on the Abi provided when the contract was [declared](https://github.com/BootNodeDev/dAppBooster/blob/contract-address-return-type/src/constants/contracts/contracts.ts#L14).

### PR DESCRIPTION
**Pull request by paterson1**
Closes #224

# Description:

(Please include a summary of the changes and the related issue)

# Steps:

(Required steps to reproduce or test the fix / feature)

## Type of change:

- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [x] Enhancement
- [ ] Refactoring
- [ ] Chore

# How Has This Been Tested?

- [ ] Manual testing
- [ ] Automated tests
- [ ] Other (explain)

# Remember to check that:

- Your code follows the style guidelines of this project
- You have performed a self-review of your code
- You have commented your code in hard-to-understand areas
- You have made corresponding changes to the documentation
- Your changes generate no new warnings

# Screenshots

(Add screenshots or videos to help test this pull request)

### CODE REVIEW

**Review by fernandomg — COMMENTED** [review]
_chapeau_

**Comment by fernandomg** [comment]
@paterson1 there are some conflicts that need to be solved too

File: src/lib/wagmi/config.ts [hunk 1]
```diff
@@ -10,5 +10,5 @@ import { reactSuspenseRead } from '@/src/lib/wagmi/plugins/reactSuspenseRead'
 export default defineConfig({
   out: 'src/hooks/generated.ts',
   plugins: [reactSuspenseRead(), react()],
-  contracts: getContracts() as ContractConfig<number, undefined>[],
+  contracts: getContracts(),
```
Comment (paterson1) [inline src/lib/wagmi/config.ts:13]: @fernandomg 🙃

**Review by fernandomg — APPROVED** [review]
I took me some time to get my head around `infer`, but by using it altogether works wonderfully

**Pull request merged**